### PR TITLE
Notebooks: Fix callout dialog being cut off at the bottom of the document

### PR DESCRIPTION
**Pull request by lucyzhang929**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/14558

Show link callout dialog above the toolbar when there is not enough space below the toolbar.

Screen Shot 2021-03-05 at 12 10 56 PM
Screen Shot 2021-03-05 at 12 11 07 PM

### CODE REVIEW

File: src/sql/workbench/browser/modal/modal.ts [hunk 1]
```diff
@@ -432,6 +432,16 @@ export abstract class Modal extends Disposable implements IThemable {
 				dialogWidth = this._modalOptions.width;
 			}
 
+			if (this._modalOptions.dialogPosition === 'above') {
+				if (this._modalOptions.dialogProperties) {
+					this._modalDialog.style.left = `${this._modalOptions.dialogProperties.xPos - this._modalOptions.dialogProperties.width}px`;
+					this._modalDialog.style.top = `${this._modalOptions.dialogProperties.yPos - 235}px`;
```
Comment (chlafreniere) [inline src/sql/workbench/browser/modal/modal.ts:438]: What does 235 refer to?

File: src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts [hunk 2]
```diff
@@ -278,11 +278,12 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		const triggerHeight = triggerElement.offsetHeight;
 		const triggerWidth = triggerElement.offsetWidth;
 		const dialogProperties = { xPos: triggerPosX, yPos: triggerPosY, width: triggerWidth, height: triggerHeight };
+		const dialogPosition = window.innerHeight - triggerPosY < 250 ? 'above' : 'below';
```
Comment (chlafreniere) [inline src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts:281]: 250 is a magic number. What does it refer to? How can we make this generic?

File: src/sql/workbench/browser/modal/modal.ts (hunk 1 above)
Comment (lucyzhang929) [reply src/sql/workbench/browser/modal/modal.ts:438]: the height of the dialog.. let me try to find that property

File: src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts (hunk 2 above)
Comment (lucyzhang929) [reply src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts:281]: the height of the dialog + the arrow at the bottom of the dialog - will also look for this property

**Review by Charles-Gagnon — APPROVED** [review]

**Review by barbaravaldez — APPROVED** [review]

File: src/sql/workbench/browser/modal/modal.ts (hunk 1 above)
Comment (lucyzhang929) [reply src/sql/workbench/browser/modal/modal.ts:438]: The height of the dialog is not stored anywhere right now - only the max-height is stored:
https://github.com/microsoft/azuredatastudio/blob/b1daa4ee4810393b2d03583f646400c2c1695149/src/sql/workbench/browser/modal/media/calloutDialog.css#L12
Given that @halerankin will be refactoring the callout code, is it alright if I leave this for now?

**Pull request merged**